### PR TITLE
Add more linter

### PR DESCRIPTION
**Pull request by PierreF**
Add black, codespell and mypy.

mypy is only partially added, mostly to client.py.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -7,6 +7,11 @@
 # default_language_version:
 #   python: python3
 repos:
+  - repo: https://github.com/psf/black
+    rev: 23.12.1
+    hooks:
+      - id: black
+
```
Comment (akx) [inline .pre-commit-config.yaml:14]: Please consider `ruff-format` instead.

I think the examples should be left alone for now too. 

If you are planning on an auto-formatting commit, please also add the Ruff `COM` rule there, and make sure not to squash-commit the PR including it.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -37,3 +42,10 @@ repos:
     rev: v0.15
     hooks:
       - id: validate-pyproject
+
+  - repo: https://github.com/codespell-project/codespell
+    rev: v2.2.6
+    hooks:
+    - id: codespell
+      additional_dependencies:
+        - tomli
```
Comment (akx) [inline .pre-commit-config.yaml:51]: Please consider [`typos`](https://github.com/crate-ci/typos) instead.

File: src/paho/mqtt/client.py [hunk 3]
```diff
@@ -131,25 +161,49 @@
...
-MQTT_ERR_TLS = 8
-MQTT_ERR_PAYLOAD_SIZE = 9
-MQTT_ERR_NOT_SUPPORTED = 10
-MQTT_ERR_AUTH = 11
-MQTT_ERR_ACL_DENIED = 12
-MQTT_ERR_UNKNOWN = 13
-MQTT_ERR_ERRNO = 14
-MQTT_ERR_QUEUE_SIZE = 15
-MQTT_ERR_KEEPALIVE = 16
+class MQTTErrorCode(enum.IntEnum):
```
Comment (akx) [inline src/paho/mqtt/client.py:166]: This shouldn't be in an "add more linter" PR!

**Review by akx — CHANGES_REQUESTED** [review]
Beyond the inline comments, I'd urge you to separate this into multiple PRs, not a single unreviewable "add more linter" one. I'm happy to review smaller PRs!

File: .pre-commit-config.yaml (hunk 1 above)
Comment (PierreF) [reply .pre-commit-config.yaml:14]: Added formatting in #793 with ruff-format

File: .pre-commit-config.yaml (hunk 2 above)
Comment (PierreF) [reply .pre-commit-config.yaml:51]: Added typos in #792

File: src/paho/mqtt/client.py (hunk 3 above)
Comment (PierreF) [reply src/paho/mqtt/client.py:166]: This was for typing fix, now seperated in #791

**Comment by PierreF** [comment]
@akx thank for your feedback.

I've splitted the PR in multiple PRs:
* #791 
* #792 
* #793

Closing this PR in favor of other one.

**Comment by akx** [comment]
@PierreF Thank you, and sorry if my comments sounded a bit harsh – I've just seen the maintenance pain of big PRs like this just steamrolled through 😅

**Pull request closed without merging**